velox/exec/Task.cpp

```diff
@@ -614,52 +635,62 @@ void Task::start(
       }
 
       self->validateGroupedExecutionLeafNodes();
-    }
 
-    // Register self for possible memory recovery callback. Do this
-    // after sizing 'drivers_' but before starting the
-    // Drivers. 'drivers_' can be read by memory recovery or
-    // cancellation while Drivers are being made, so the array should
-    // have final size from the start.
-    auto bufferManager = self->bufferManager_.lock();
-    VELOX_CHECK_NOT_NULL(
-        bufferManager,
-        "Unable to initialize task. "
-        "PartitionedOutputBufferManager was already destructed");
+      // Register self for possible memory recovery callback. Do this
```

Comment:
Not your code, but I don't see what this comment refers to.

> so the array should have final size from the start.

Which array is this about?